### ISSUE TITLE
sample not working

### ISSUE THREAD

**Issue by iverwei** [Azure-Samples/active-directory-b2c-ios-native-appauth#3]
I download this sample code and change Endpoint/ClientId...etc variables to match my AAD B2C tenant/app id, but I got error message after I successfully signed in to Facebook.

the error message is Authorization error: The operation couldn’t be completed. (org.openid.appauth.general error -4.)

Authorize endpoint: https://login.microsoftonline.com/te/sksb2c.onmicrosoft.com/B2C_1_sksb2cuserweb_sign_up_in/oauth2/v2.0/authorize

Token endpoint: https://login.microsoftonline.com/te/sksb2c.onmicrosoft.com/B2C_1_sksb2cuserweb_sign_up_in/oauth2/v2.0/token

**Comment by parakhj** [comment]
Could you share more about which values you changed? The only values that you need to change are the following values in ViewController.m

```
static NSString *const kTenantName = @"fabrikamb2c.onmicrosoft.com";
static NSString *const kSignupOrSigninPolicy = @"b2c_1_susi";
static NSString *const kEditProfilePolicy = @"b2c_1_edit_profile";
static NSString *const kClientId = @"90c0fe63-bcf2-44d5-8fb7-b8bbc0b29dc6";
static NSString *const kRedirectUri = @"com.onmicrosoft.fabrikamb2c.exampleapp://oauth/redirect";
```

and the following:

Open Info.pList. Expand 'URL types' -> 'Item 0' -> 'URL Schemes' -> 'Item 0'. Update the scheme 'com.onmicrosoft.fabrikamb2c.exampleapp' to match the scheme of the kRedirectUri value updated in ViewController.m.

**Comment by gsacavdm** [comment]
Also, can you confirm if the sample worked before you made changes with the out-of-the-box configuration?

**Comment by iverwei** [comment]
@parakhj 
Hi,I changed the value for  kTenantName, kSignupOrSigninPolicy, kClientId and kRedirectUri, 
and I checked info.plist schemes value and I am sure it is the same as the value in ViewController.m  kRedirectUri.

@gsacavdm 
Hi,
 
The code works fine ‘out-of-the-box’ when I download the code.
 
But then I change the value for  kTenantName, kSignupOrSigninPolicy, kClientId and kRedirectUri, and I checked info.plist schemes value and I am sure it is the same as the value in ViewController.m  kRedirectUri.
 
Not sure I have missed something things somewhere?
 
Or if it’s possible for me to send you my code via email?
 
Thanks!

**Comment by parakhj** [comment]
"I am sure it is the same as the value in ViewController.m kRedirectUri." 

Just to confirm, it shouldn't be the exact same value. It should just be the first part of the redirect URI ('com.onmicrosoft.fabrikamb2c.exampleapp') not "com.onmicrosoft.fabrikamb2c.exampleapp://oauth/redirect"

**Comment by schokolateur** [comment]
Hi there, 

the sample is also not working for me.
First, the pod cannot be build, don't know why. My system is up to date (Xcode and so on).
I put the source of AppAuth directly into the project. 
Now i can build an run.
I changed the data as described, but when the login redirects, it redirects to localhost and safari cannot open the page...
Am i right, that i have to use the custom redirect uri field? 
I have tested in simulator and on my phone.
So do you have any hits for me?
Thanks

**Comment by parakhj** [comment]
Yes, you need to use the custom redirect uri. Did you get it working?

**Comment by alvincrisuy** [comment]
Did anyone have an update for this? I also encounter this issue

**Comment by MananKhatri** [comment]
> I download this sample code and change Endpoint/ClientId...etc variables to match my AAD B2C tenant/app id, but I got error message after I successfully signed in to Facebook.
> 
> the error message is Authorization error: The operation couldn’t be completed. (org.openid.appauth.general error -4.)
> 
> Authorize endpoint: https://login.microsoftonline.com/te/sksb2c.onmicrosoft.com/B2C_1_sksb2cuserweb_sign_up_in/oauth2/v2.0/authorize
> 
> Token endpoint: https://login.microsoftonline.com/te/sksb2c.onmicrosoft.com/B2C_1_sksb2cuserweb_sign_up_in/oauth2/v2.0/token

did your problem solve?
